### DISCUSSION TITLE
如何监听当x-component是自定义组件的值变化 ？

### DISCUSSION THREAD

**Discussion by wangjie19910615** [alibaba/formily#1879] [Q&A 2.x] [answered]
如何监听当x-component是自定义组件的值变化

**Comment by janryWang** [comment]
没懂，描述清楚一点

**Reply by wangjie19910615** [reply]
补充了codesandbox，自定义组件回显数据的时候，如何动态更新联动数据输入框的数据
https://codesandbox.io/s/upbeat-glade-r2mug?file=/App.tsx #1879

**Reply by janryWang** [reply]
还是没看懂你这个例子。。。
更新输入框数据，直接设置字段的value不就行了么

**Reply by wangjie19910615** [reply]
dataSource: {
      type: "string",
      title: "联动数据",
      "x-decorator": "FormItem",
      "x-component": "Input",
      "x-reactions": {
        dependencies: "deviceId",
        fulfill: {
          //run: "test($self, $form.getFieldState('deviceId'))",
          state: {
            value: "{{$form.getFieldState('deviceId').value}}"
          }
        }
      }
    },

x-reactions没有监控到deviceId的变化

**Reply by janryWang** [reply]
复现不了，$form.getFieldState('deviceId')?.value本身就能监听变化

**Comment by janryWang** [comment, marked as answer]
自定义组件，如果你是自己的业务组件，observer就够了，不要用connect，connect只是对于第三方组件做无侵入接入用的

**Reply by wangjie19910615** [reply]
你好，在自定义组件中使用form.setValue是可以的，想请问如果其他组件想要订阅自定义组件内的值变化，应该怎么做？

**Reply by janryWang** [reply]
你是没看过文档？先好好看文档

**Reply by wangjie19910615** [reply]
我是把输入框作为自定义组件，然后form-》field->value一直是undefined，所以监控不到值变化

**Reply by janryWang** [reply]
我只能说formily antd提供的所有组件，都是自定义组件，没有任何特殊区分

**Reply by wangjie19910615** [reply]
解决了，感谢。原来是没有在自定义组件中使用useField设置value
